src/shared/openssl-util.c

```diff
@@ -1418,3 +1403,34 @@ int x509_fingerprint(X509 *cert, uint8_t buffer[static SHA256_DIGEST_SIZE]) {
         return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "OpenSSL is not supported, cannot calculate X509 fingerprint: %m");
 #endif
 }
+
+int parse_openssl_key_source_argument(
+                const char *argument,
+                char **ret_private_key_source,
+                OpensslKeySourceType *ret_private_key_source_type) {
```

Comment:
please drop the `ret_` prefix from these parameters, as they are both input *and* output (correctly so)